src/components/table/table-story-helpers/table-story.hoc.js

```diff
@@ -0,0 +1,140 @@
+import { text } from '@storybook/addon-knobs';
+import React, { useState } from 'react';
+import { ThemeProvider } from 'styled-components';
+import { action } from '@storybook/addon-actions';
+import { State, Store } from '@sambego/storybook-state';
+import { commonKnobs } from './table-story-knobs';
+import countriesList from '../../../../demo/data/countries';
+import Button from '../../button';
+import MultiActionButton from '../../multi-action-button/multi-action-button';
+import {
+  Table, TableCell, TableHeader, TableRow
+} from '..';
+
+const store = new Store({
+  sortOrder: commonKnobs().sortOrder,
+  sortedColumn: commonKnobs().sortedColumn,
+  currentPage: '1',
+  children: undefined
+});
+
+const handleChange = (e, tableOptions) => {
+  const { sortOrder, sortedColumn, currentPage } = tableOptions;
+
+  store.set({ sortOrder, sortedColumn, currentPage });
+  action('change')(e, tableOptions);
+};
+
+const recordsForActivePage = (start, end) => {
+  let records = countriesList;
+  if (store.get('sortOrder') === 'desc' && store.get('sortedColumn').length) {
+    records = records.reverse();
+  }
+  return records.slice(start, end).toJS();
+};
+
+const getActiveRows = (pageSize, totalRecords) => {
+  const currentPage = store.get('currentPage');
+  const candidateIndex = pageSize * currentPage;
+
+  const endIndex = (candidateIndex <= totalRecords) ? candidateIndex : totalRecords;
+  const currentPageSize = (endIndex === totalRecords) ? (endIndex % pageSize) : pageSize;
+  const startIndex = endIndex - currentPageSize;
+
+  return recordsForActivePage(startIndex, endIndex);
+};
+
+const buildRows = ({ pageSize, totalRecords }) => {
+  const rowsCountries = getActiveRows(pageSize, totalRecords);
+
+  return (
+    <>
+      <TableRow
+        key='header'
+        as='header'
+        uniqueID='header'
+      >
+        <TableHeader
+          sortable
+          name='name'
+          scope='col'
+        >
+        Country
+        </TableHeader>
+        <TableHeader
+          sortable
+          scope='col'
+          name='code'
+          width='200'
+        >
+        Code
+        </TableHeader>
+      </TableRow>
+      {rowsCountries.map(row => (
+        <TableRow
+          key={ row.id }
+          uniqueID={ row.id }
+        >
+          <TableCell>{row.name}</TableCell>
+          <TableCell>{row.value}</TableCell>
+        </TableRow>
+      ))}
+  </>
+  );
+};
+
+const setCurrentPage = ({ totalRecords, pageSize, paginate }) => {
```

Comment:
The name `setCurrentPage` seems a bit misleading here – it sounds like executing the function will actually set the current page, which it doesn't.

Maybe a name like `calculateCurrentPage` would be better? (since all it's doing is calculating and returning the current page number, based on the supplied `tableProps` and the `store.get('currentPage')` value)

(I realise this isn't new code – it's simply been moved from `src/components/table/table.stories.js` – but maybe we could take this opportunity to improve the name?)

Reply:
Yeah good point I'll make the change 👍

Reply:
@edleeks87: Great, many thanks :white_check_mark: